packages/quantic/force-app/main/default/lwc/quanticSmartSnippet/quanticSmartSnippet.js

```diff
@@ -160,6 +176,89 @@ export default class QuanticSmartSnippet extends LightningElement {
     }
   }
 
+  /**
+   * Handles the "like" button press.
+   * @returns {void}
+   */
+  like = (event) => {
+    event.stopPropagation();
+    if (!this.liked) {
+      this.smartSnippet.like();
+      this.hideExplainWhyFeedbackButton = true;
+    }
+  };
+
+  /**
+   * Handles the "dislike" button press.
+   * @returns {void}
+   */
+  dislike = (event) => {
+    event.stopPropagation();
+    if (!this.disliked) {
+      this.smartSnippet.dislike();
+      if (!this.feedbackSubmitted) {
+        this.hideExplainWhyFeedbackButton = false;
+      }
+    }
+  };
+
+  /**
+   * Handles the "explain why" button press.
+   * @returns {void}
+   */
+  explainWhy = (event) => {
+    event.stopPropagation();
+    this.openFeedbackModal();
+  };
+
+  /**
+   * Opens the feedback modal.
+   */
+  async openFeedbackModal() {
+    this.smartSnippet.openFeedbackModal();
+    await FeedbackModal.open({
+      label: 'Explain why',
+      size: 'small',
+      description: 'Feedback modal',
+      options: this.options,
+      submitFeedback: this.submitFeedback.bind(this),
+    });
+    this.smartSnippet.closeFeedbackModal();
+  }
+
+  /**
+   * Submits the feedback
+   * @param {{value: SmartSnippetFeedback, details: string}} feedback
+   * @returns {void}
+   */
+  submitFeedback(feedback) {
+    if (feedback?.details) {
+      this.smartSnippet.sendDetailedFeedback(feedback.details);
+    } else if (feedback?.value) {
+      this.smartSnippet.sendFeedback(feedback.value);
+    }
+    this.feedbackSubmitted = true;
+    this.hideExplainWhyFeedbackButton = true;
+  }
+
+  get options() {
```

Comment:
I'm wondering whether setting options this way is too rigid for customers.

Suppose a customer wants different options, then he has to duplicate the `quanticSmartSnippet` component just to override the options.

Suppose a customer wants the detail textarea for more than one option, then he has to duplicate and modify the `quanticFeebackModal`, `quanticFeedback`, and `quanticSmartSnippet` components.

I think making this component more modular could go a long way for customers implementing Smart Snippets.

Reply:
The reason why I'm setting the options this way is because in Headless only these predefined options are allowed in the analytics sent when submitting a feedback.
For instance the `sendFeedback` method of the Headless Smart Snippet controller can only be called with a parameter of type `SmartSnippetFeedback ` which is a type defined in Headless as: 
```type SmartSnippetFeedback = "does_not_answer" | "partially_answers" | "was_not_a_question"```

For this reason I didn't consider adding the ability to accept `options` as a public property for example.
If a customer wants different options, then he has to duplicate the quanticSmartSnippet component NOT just to override the options but also to handle how the feedback will be submitted since in that particular case the methods that come with the Headless Smart Snippet controller will be no help for this customer.

Basically  we can't offer a lot of freedom here just because we currently don't have the same freedom in the Headless Smart Snippet controller.

Reply:
I hear you and this makes perfect sense to me as well!
I think this would need to be in a separate PR just because this addition would require a change in Headless as well: 
we will need to change the signature of the function `sendFeedback` from:
```
function sendFeedback(feedback: SmartSnippetFeedback)
```
to: 
```
function sendFeedback(feedback: SmartSnippetFeedback, details?: string)
```
After doing this change in Headless, we will be able to offer more freedom to our customers in Quantic when it comes to giving smart snippet feedbacks.

Does this sound right to you?

Reply:
It seems like a good plan to me